sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/util/InternalContextN.java

```diff
@@ -71,12 +74,12 @@ public InternalContext put(Object key, Object value) {
 
     @Override
     public Object get(Object key) {
-        Object data = null;
+        Object data = SENTINEL;
 
         // Iterate in reverse order to get the most recent data first.
         for (int i = contexts.length - 1; i >= 0; i--) {
             data = contexts[i].get(key);
-            if (data != null) {
+            if (!Objects.equals(SENTINEL, data)) {
```

Comment:
Wouldn't `Objects.equals(SENTINEL, null)` return false?

Reply:
Correct, this is to allow `null` values to be set in Context and returned properly. If we had

- key: foo
\ - key2: bar
\ - key: null

We'd want null to be returned as it is last in the chain